### PR TITLE
attribute inference for destruction

### PR DESCRIPTION
**Pull request by radcapricorn**

### CODE REVIEW

File: source/automem/utils.d [hunk 1]
```diff
@@ -1,55 +1,90 @@
...
 
-@("hasNoGcDestructor")
-@safe pure unittest {
-    static assert(hasNoGcDestructor!NoGc);
-    static assert(!hasNoGcDestructor!Gc);
+template _isStaticArray(T) {
+    enum bool _isStaticArray = false;
 }
 
+template _Seq(TList...) {
```
Comment (atilaneves) [inline source/automem/utils.d]: Isn't this the same as `AliasSeq`?

File: source/automem/utils.d [hunk 2]
```diff
@@ -1,55 +1,90 @@
...
+void destruct(T)(ref T obj)
+if(!is(T == struct) && !is(T == class) && !is(T == interface) && !_isStaticArray!T) {
+    obj = T.init;
 }
 
-// enum hasNoGcDestructor(T) = is(T == class) &&
-//     (!hasMember!(T, "__dtor") || (functionAttributes!(typeof(T.__dtor)) & FunctionAttribute.nogc));
+private:
+
+template _isStaticArray(T : U[n], U, size_t n) {
```
Comment (atilaneves) [inline source/automem/utils.d]: std.traits?

File: source/automem/utils.d (hunk 1 above)
Comment (radcapricorn) [reply source/automem/utils.d]: Yeah, it is. I basically copy-pasted it from my druntime sandbox, which in turn was a copypaste from druntime master.

Comment (atilaneves) [reply source/automem/utils.d]: So why not use `AliasSeq`?

Comment (radcapricorn) [reply source/automem/utils.d]: If you don't mind std. dependencies there, I'll change those to use std.traits.

Comment (atilaneves) [reply source/automem/utils.d]: `std` is preferable. I'd only possibly be bothered by dependencies on other dub packages.

**Pull request merged**